models/HealthCheck.ts

```diff
@@ -4,38 +4,44 @@ import { MonitoredService, ServiceHealthCheckResponse } from '../types';
 
 export interface HealthCheckResults {
   healthStatus: string;
-  failedHealthChecks: Array<ServiceHealthCheckResponse>;
+  failedHealthChecks: ServiceHealthCheckResponse[];
 }
 
 export default class HealthCheck {
-  private healthCheckResults: HealthCheckResults
-  private services: MonitoredService[]
+  private healthCheckResults: HealthCheckResults;
 
-  constructor(services: MonitoredService[]) {
-    this.healthCheckResults = { healthStatus: 'vibrant', failedHealthChecks: [] };
+  private readonly services: MonitoredService[];
+
+  public constructor(services: MonitoredService[]) {
+    this.healthCheckResults = { failedHealthChecks: [], healthStatus: 'vibrant' };
     this.services = services;
   }
 
   public async check(): Promise<void> {
-    const resultPromises: Promise<ServiceHealthCheckResponse>[] =
-      this.services.map(service => service.healthCheck());
+    const resultPromises: Array<Promise<ServiceHealthCheckResponse>> = this.services.map(service =>
```

Comment:
How come resultPromises is `Array<Promise<ServiceHealthCheckResponse>>` but failedHealthChecks got changed to `ServiceHealthCheckResponse[]`?

Reply:
For `failedHealthChecks: Array<ServiceHealthCheckResponse>` - `Array type using 'Array<ServiceHealthCheckResponse>' is forbidden for simple types. Use 'ServiceHealthCheckResponse[]' instead.` 

For `const resultPromises: Promise<ServiceHealthCheckResponse>[]` - `Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.`

Reply:
These are both configurable in the eslint configuration if we felt like changing them. We would probably need to discuss with folks why they set up the rules like that in the first place for the frontend before we change them here.

Reply:
The Promise is the culprit I guess. Thanks for helping me understand!

Reply:
👍  Yeah, I'm thinking the `Promise<T>` makes it "non-simple"